addons/l10n_dk/models/account_move.py

```diff
@@ -0,0 +1,21 @@
+from odoo import api, fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    l10n_dk_currency_rate_at_transaction = fields.Float(
+        string='Rate',
+        compute='_compute_currency_rate_at_transaction', readonly=True, store=True,
+        digits=(12, 6),
+    )
+
+    @api.depends('currency_id', 'invoice_date', 'date', 'company_id')
+    def _compute_currency_rate_at_transaction(self):
+        for record in self:
+            record.l10n_dk_currency_rate_at_transaction = self.env['res.currency']._get_conversion_rate(
```

Comment:
Might be better to take the same rate as the one from the lines? `line_ids.currency_rate`